pydm/widgets/enum_button.py

```diff
@@ -127,6 +130,158 @@ def orientation(self, new_orientation):
             self._orientation = new_orientation
             self.rebuild_layout()
 
+    @Property(int)
+    def margin_top(self):
+        """
+        The top margin of the QGridLayout of buttons.
+
+        Returns
+        -------
+        int
+        """
+        return self._layout_margins.top()
+
+    @margin_top.setter
+    def margin_top(self, new_margin):
+        """
+        Set the top margin of the QGridLayout of buttons.
+
+        Parameters
+        -------
+        int
+        """
+        new_margin = max(0, int(new_margin))
+        self._layout_margins.setTop(new_margin)
+        self.layout().setContentsMargins(self._layout_margins)
+        self.rebuild_layout()
+
+    @Property(int)
+    def margin_bottom(self):
+        """
+        The bottom margin of the QGridLayout of buttons.
+
+        Returns
+        -------
+        int
+        """
+        return self._layout_margins.bottom()
+
+    @margin_bottom.setter
+    def margin_bottom(self, new_margin):
+        """
+        Set the bottom margin of the QGridLayout of buttons.
+
+        Parameters
+        -------
+        int
+        """
+        new_margin = max(0, int(new_margin))
+        self._layout_margins.setBottom(new_margin)
+        self.layout().setContentsMargins(self._layout_margins)
+        self.rebuild_layout()
```

Comment:
It is not necessary to `rebuild_layout` since you are already applying the new value to the layout.
```suggestion

```